test/bigtable/filter.js

```diff
@@ -0,0 +1,677 @@
+
+    mockery.enable({
+      useCleanCache: true,
+      warnOnUnregistered: false
+    });
+
+    Filter = require('../../lib/bigtable/filter');
+  });
+
+  after(function() {
+    mockery.deregisterAll();
+    mockery.disable();
+  });
+
+  beforeEach(function() {
+    filter = new Filter();
+  });
+
+  afterEach(function() {
+    spies.forEach(function(spy) {
+      spy.calls.length = 0;
+    });
+  });
+
+  describe('instantiation', function() {
+    it('should create an empty array of filters', function() {
+      assert.deepEqual(filter.filters_, []);
+    });
+  });
+
+  describe('convertToRegExpString', function() {
+    it('should convert a RegExp to a string', function() {
+      var str = Filter.convertToRegExpString(/\d+/);
+
+      assert.strictEqual(str, '\\d+');
+    });
+
+    it('should convert an Array of strings to a single string', function() {
+      var things = ['a', 'b', 'c'];
+      var str = Filter.convertToRegExpString(things);
+
+      assert.strictEqual(str, '(a|b|c)');
+    });
+
+    it('should not do anything to a string', function() {
+      var str1 = 'hello';
+      var str2 = Filter.convertToRegExpString(str1);
+
+      assert.strictEqual(str1, str2);
+    });
+
+    it('should convert a number to a string', function() {
+      var str = Filter.convertToRegExpString(1);
+
+      assert.strictEqual(str, '1');
+    });
+
+    it('should throw an error for unknown types', function() {
+      var errorMessage = /Can\'t convert to RegExp String from unknown type/;
+
+      assert.throws(function() {
+        Filter.convertToRegExpString(true);
+      }, errorMessage);
+    });
+  });
+
+  describe('createRange', function() {
+    it('should create a range object', function() {
+      var start = 'a';
+      var end = 'b';
+      var key = 'Key';
+
+      var range = Filter.createRange(start, end, key);
+
+      assert.deepEqual(range, {
+        startKeyInclusive: start,
+        endKeyInclusive: end
+      });
+    });
+
+    it('should only create start bound', function() {
+      var start = 'a';
+      var key = 'Key';
+
+      var range = Filter.createRange(start, null, key);
+
+      assert.deepEqual(range, {
+        startKeyInclusive: start
+      });
+    });
+
+    it('should only create an end bound', function() {
+      var end = 'b';
+      var key = 'Key';
+
+      var range = Filter.createRange(null, end, key);
+
+      assert.deepEqual(range, {
+        endKeyInclusive: end
+      });
+    });
+
+    it('should optionally accept inclusive flags', function() {
+      var start = {
+        value: 'a',
+        inclusive: false
+      };
+
+      var end = {
+        value: 'b',
+        inclusive: false
+      };
+
+      var key = 'Key';
+
+      var range = Filter.createRange(start, end, key);
+
+      assert.deepEqual(range, {
+        startKeyExclusive: start.value,
+        endKeyExclusive: end.value
+      });
+    });
+  });
+
+  describe('parse', function() {
```

Comment:
From here down, I believe each method tested needs a test to check the value being returned from the call to the function itself. e.g. some return `this`, others return the result of `this.set()`.